src/main/java/net/minecraftforge/fmp/microblock/MicroblockDelegate.java

```diff
@@ -0,0 +1,175 @@
+package net.minecraftforge.fmp.microblock;
+
+import com.google.common.base.Optional;
+
+import net.minecraft.block.Block;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraftforge.fmp.multipart.IMultipart;
+import net.minecraftforge.fmp.multipart.Multipart;
+
+/**
+ * Represents an object to which microblock events are delegated. This allows for custom interaction and data storage
+ * inside the microblock.
+ */
+public class MicroblockDelegate
```

Comment:
You can extend that class to add custom functionality to your microblock materials. One example would be Framez's Metamorphic Stone, which turns fluid blocks into solids: https://gist.github.com/amadornes/d708687c67f35b45dd92a75c7fb80606
A delegate provider is registered along with the microblock material and whenever a new microblock is created, if there's a delegate for that material, it's created.